Development/nmos/client_utils.cpp

```diff
@@ -55,39 +60,165 @@ namespace nmos
+        {
+            utility::string_t interface_name;
+            if (!address.empty())
+            {
+                const auto interfaces = web::hosts::experimental::host_interfaces();
+
+                auto find_interface = [&]()
+                {
+                    return boost::range::find_if(interfaces, [&](const web::hosts::experimental::host_interface& interface)
+                    {
+                        return interface.addresses.end() != boost::range::find(interface.addresses, address);
+                    });
+                };
+                const auto interface = find_interface();
+                if (interfaces.end() != interface) { interface_name = interface->name; }
+            }
+            return interface_name;
+        }
+
+#if !defined(_WIN32) || !defined(__cplusplus_winrt) || defined(CPPREST_FORCE_HTTP_CLIENT_ASIO)
+        // bind socket to a specific network interface, only supporting Linux
+#if defined(linux) || defined(__linux) || defined(__linux__)
+        inline bool bind_to_device(const utility::string_t& interface_name, bool secure, void* native_handle)
+        {
+            int socket_fd;
+            if (secure)
+            {
+                auto socket = (boost::asio::ssl::stream<boost::asio::ip::tcp::socket&>*)native_handle;
+                if (!socket->lowest_layer().is_open())
+                {
+                    // for now, limited to IPv4
+                    socket->lowest_layer().open(boost::asio::ip::tcp::v4());
+                }
+                socket_fd = socket->lowest_layer().native_handle();
+            }
+            else
+            {
+                auto socket = (boost::asio::ip::tcp::socket*)native_handle;
+                if (!socket->is_open())
+                {
+                    // for now, limited to IPv4
+                    socket->open(boost::asio::ip::tcp::v4());
+                }
+                socket_fd = socket->lowest_layer().native_handle();
+            }
+            // SO_BINDTODEVICE not defined in windows & mac
+            return (setsockopt(socket_fd, SOL_SOCKET, SO_BINDTODEVICE, utility::us2s(interface_name).c_str(), interface_name.length()) == 0);
+        }
+#endif
+
+        inline std::function<void(web::http::client::native_handle)> make_client_nativehandle_options(const utility::string_t& client_address, bool secure, slog::base_gate& gate)
+        {
+            // get the associated network interface name from IP address
+            const auto interface_name = get_interface_name(client_address);
+            if (!client_address.empty() && interface_name.empty())
+            {
+                slog::log<slog::severities::error>(gate, SLOG_FLF) << "No network interface found for " << client_address << " to bind for the HTTP client connection";
+            }
+
+            return [interface_name, secure, &gate](web::http::client::native_handle native_handle)
+            {
+                if (!interface_name.empty())
```

Comment:
Why not just immediately return an empty or no-op function from `make_client_nativehandle_options` if `client_address` is empty (or handle this outside this function completely)?

Reply:
Okay, add no-op function in event of an empty interface name